addons/sourcemod/scripting/shavit-tas.sp

```diff
@@ -0,0 +1,1395 @@
+		int iFrameNumber = gI_IndexCounter[client];
+		if(iFrameSize > 1 && iFrameNumber > 2)
+		{
+			float fAngle[3];
+			fAngle[0] = GetArrayCell(gA_Frames[client], iFrameNumber, 3);
+			fAngle[1] = GetArrayCell(gA_Frames[client], iFrameNumber, 4);
+			fAngle[2] = 0.0;
+			
+			float fPosition2[3];
+			fPosition2[0] = GetArrayCell(gA_Frames[client], iFrameNumber, 0);
+			fPosition2[1] = GetArrayCell(gA_Frames[client], iFrameNumber, 1);
+			fPosition2[2] = GetArrayCell(gA_Frames[client], iFrameNumber, 2);
+
+			float fPosition[3];
+			fPosition[0] = GetArrayCell(gA_Frames[client], iFrameNumber-1, 0);
+			fPosition[1] = GetArrayCell(gA_Frames[client], iFrameNumber-1, 1);
+			fPosition[2] = GetArrayCell(gA_Frames[client], iFrameNumber-1, 2);
+
+			float fVelocity[3];
+			MakeVectorFromPoints(fPosition2, fPosition, fVelocity);
+
+			for (int i = 0; i < 3; i++)
+			{
+				fVelocity[i] *= RoundToFloor(gF_TickRate);
+			}
+
+			TeleportEntity(client, fPosition, fAngle, fVelocity);
+
+			if(GetArrayCell(gA_Frames[client], iFrameNumber, 5) & IN_DUCK)
+			{
+				SetEntProp(client, Prop_Send, "m_bDucked", true);
+				SetEntProp(client, Prop_Send, "m_bDucking", true);
+			}
+			else
+			{
+				SetEntProp(client, Prop_Send, "m_bDucked", false);
+				SetEntProp(client, Prop_Send, "m_bDucking", false);
+			}
+
+			gF_IndexCounter[client] -= gF_CounterSpeed[client];
+			if(IsRound(gF_IndexCounter[client]))
+			{
+				gI_IndexCounter[client]--;
+			}
+			gF_TASTime[client] -= GetTickInterval() * gF_CounterSpeed[client];
+		}
+	}
+	else
+	{ //TAS Status must be PAUSED to use this function
+		Shavit_PrintToChat(client, "sm_minusone can only be used when TAS is Paused!");
+	}
+	return Plugin_Handled;
+}
+
+public Action Command_TASMenu(int client, int args)
+{
+	gB_TASMenu[client] = !gB_TASMenu[client];
+	if(gB_TASMenu[client])
+	{
+		DrawPanel(client);
+	}
+	return Plugin_Handled;
+}
+
+public Action Command_TASHelp(int client, int args)
+{
+	Shavit_PrintToChat(client, "TAS Guide:");
+	Shavit_PrintToChat(client, "Recommended Binds:");
+	Shavit_PrintToChat(client, "bind mwheelup sm_minusone");
+	Shavit_PrintToChat(client, "bind mwheeldown sm_plusone");
+	Shavit_PrintToChat(client, "bind mouse1 +rewind");
+	Shavit_PrintToChat(client, "bind mouse2 +fastforward");
+	Shavit_PrintToChat(client, "Other Commands:");
+	Shavit_PrintToChat(client, "+autostrafer - When bound hold to use wigglehack");
+	Shavit_PrintToChat(client, "!tasmenu - Toggles TAS Menu");
+	return Plugin_Handled;
+}
+
+public Action CommandListener_TAS(int client, const char[] command, int args)
+{
+	gB_TASMenu[client] = true;
+	DrawPanel(client);
+	return Plugin_Continue;
+}
+
+public Action CommandListener_JoinTeam(int client, const char[] command, int args)
+{
+	if(gB_TAS[client])
+	{
+		gI_Status[client] = RUN;
+		gF_TASTime[client] = 0.0;
+		gI_IndexCounter[client] = 0;
+		ClearArray(gA_Frames[client]);
+		FakeClientCommandEx(client, "sm_r");
+	}
+	return Plugin_Continue;
+}
```

Comment:
should be removed. there's no reason to reset stats on team change or to force a player to restart on team change.